readthedocs/proxito/tests/test_hosting.py

```diff
@@ -766,7 +766,7 @@ def test_number_of_queries_url_subproject(self):
                 active=True,
             )
 
-        with self.assertNumQueries(24):
+        with self.assertNumQueries(28):
```

Comment:
This is a pretty large increase in queries.. is this because of the URL generation?

Reply:
Yes, it's the only thing I added in this PR.

I suppose this is because we are accessing related objects from the original object (e.g. `obj.project.slug` where `obj` is a `Build` instance). This could improve if we tell the serializer to do a prefect of these objects, I guess.

I will give it a quick try and see what I find.

Reply:
I was able to reduce 2 queries by using `.select_related()` on the objects we use all over the view.

_However_, the rest of the DB queries come from the resolver. This is because we are re-instanciating the `Resolver()` each time we want to build a URL. Example:

https://github.com/readthedocs/readthedocs.org/blob/79f1795cf50c8e55ec1eeb5c492db9e5398a2f1d/readthedocs/projects/models.py#L652-L663

At some point in the past, I suggested to create a "global `Resolver()` instance per-request" and share it across the whole request. This will reduce the amount of DB queries related to the resolver a lot. This PR is related to what I'm mentioning here.

We don't need to do this optimization _right now_, but it's something to keep in mind.